### ISSUE TITLE
MACOSX_DEPLOYMENT_TARGET inconsistent with the version of the SDK actually used.

### ISSUE THREAD

**Issue by SylvainCorlay** [scikit-build/scikit-build#480]
The `MACOSX_DEPLOYMENT_TARGET` environment variable is properly taken into account when tagging the wheel and naming the `.whl` file appropriately.

However, it does not seem to have an impact with the actual SDK that the wheel is linked with. On my OSX 10.13 machine, the 10.13 SDK is picked up.

**Example:**

```
MACOSX_DEPLOYMENT_TARGET=10.9 /Users/scorlay/miniconda3/bin/pip wheel . --verbose
```

produces **`xeus_python-0.7.1-cp37-cp37m-macosx_10_9_x86_64.whl`** as expected, but:

```
otool -l xpython_extension.cpython-37m-darwin.so | grep sdk
```

shows **`sdk 10.14`**.

However, the `CMAKE_OSX_DEPLOYMENT_TARGET` variable has the correct value (10.9) when printed by cmake.

**Current workaround**

We worked around the issue by setting `CMAKE_OSX_SYSROOT` to a path containing the correct version of the sysroot. I think that cmake should complain if it does not find a version of the SDK that is consistent with CMAKE_OSX_DEPLOYMENT_TARGET.

**Comment by SylvainCorlay** [comment]
Reading from the documentation, it seems that scikit-build forcibly sets `CMAKE_OSX_DEPLOYMENT_TARGET` and `CMAKE_OSX_ARCHITECTURES`, getting the value from `skbuild_plat_name()`, but does not use the `MACOSX_DEPLOYMENT_TARGET` environment variable for the former.

**Comment by SylvainCorlay** [comment]
I have updated the description of the issue to be more accurate.

**Comment by SylvainCorlay** [comment]
It is claimed [here](https://github.com/conda-forge/conda-forge-ci-setup-feedstock/issues/67) that the actual sysroot used does not matter, and only the deployment target does.

**Comment by SylvainCorlay** [comment]
So this may actually not be a bug...

**Comment by henryiii** [comment]
This should be inline with how setuptools handles it now. It's also much better now that PyPy has corrected their build to no longer be 10.7.

The way it works is that you can use a newer SDK to target older macOS's, Apple puts protections around all runtime features.
